### PR TITLE
e2e: make wait and traffic timeout configurable

### PR DESCRIPTION
**Pull request by MustafaSaber**
Use flags to configure e2e timeouts constants with default old values.
Signed-off-by: Mustafa Abdelrahman <mustafa.abdelrahman@zalando.de>

### CODE REVIEW

File: cmd/e2e/test_environment.go [hunk 1]
```diff
@@ -25,6 +27,8 @@ var (
 	clusterDomainInternal                              = requiredEnvar("CLUSTER_DOMAIN_INTERNAL")
 	clusterDomains                                     = []string{clusterDomain, clusterDomainInternal}
 	controllerId                                       = os.Getenv("CONTROLLER_ID")
+	waitTimeout                                        = *flag.Duration("wait-timeout", 60*time.Second, "Waiting interval before getting the resource")
+	trafficSwitchWaitTimeout                           = *flag.Duration("traffic-switch-wait-timeout", 150*time.Second, "Waiting interval before getting the checking stackset new traffic")
```
Comment (AlexanderYastrebov) [inline cmd/e2e/test_environment.go:31]: Could you check if this actually works if you chnage the value, e.g. to a small `-wait-timeout=1ms`

I think either `waitTimeout` should be a pointer
```go
waitTimeout = flag.Duration("wait-timeout", ...)
...
return &awaiter{
    t:           t,
    description: description,
    timeout:     *waitTimeout,
}
```
or should be decalred with zero value and referenced in init:
```go
var waitTimeout time.Duration
...
func init() {
    flag.DurationVar(&waitTimeout, ...)
}
```

**Comment by MustafaSaber** [comment]
That's on 150ms 
```
TEST_NAME=TestPrescalingWithHPA CLUSTER_DOMAIN=example.org CLUSTER_DOMAIN_INTERNAL=ingress.cluster.local CLUSTER_NAME=example ./e2e/run_e2e.sh -wait-timeout="150ms"
make: Nothing to be done for `build.local'.
make: `build/e2e' is up to date.
>>> Writing controller logs in /tmp/ssc-log-1662117920.log
Starting to serve on 127.0.0.1:8001
namespace/ssc-e2e-8f44c2d7701dd7ba created
=== RUN   TestPrescalingWithHPA
=== PAUSE TestPrescalingWithHPA
=== CONT  TestPrescalingWithHPA
    test_utils.go:112: Waiting 150ms for creation of stack stackset-prescale-hpa-v1...
    test_utils.go:117: Wait deadline exceeded
    prescaling_test.go:93:
        	Error Trace: /stackset-controller/prescaling_test.go:93
        	Error:      	Received unexpected error:
        	            	stacks.zalando.org "stackset-prescale-hpa-v1" not found
        	Test:       	TestPrescalingWithHPA
--- FAIL: TestPrescalingWithHPA (1.35s)
FAIL
namespace "ssc-e2e-8f44c2d7701dd7ba" deleted
```

**Comment by gargravarr** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Pull request merged**